pkg/services/ngalert/migration/alert_rule.go

```diff
@@ -68,24 +68,23 @@ func (om *OrgMigration) migrateAlert(ctx context.Context, l log.Logger, da *migr
 	titleDedupSet := om.AlertTitleDeduplicator(info.NewFolderUID)
 	name := truncate(da.Name, store.AlertDefinitionMaxTitleLength)
 	if titleDedupSet.contains(name) {
-		dedupedName := titleDedupSet.deduplicate(name)
-		l.Debug("Duplicate alert rule name detected, renaming", "oldName", name, "newName", dedupedName)
+		// This should work most of the time and will give a much better alert name than a random uid.
+		dedupedName := simpleTitleDedup(da, name)
+		if titleDedupSet.contains(dedupedName) {
+			// If we still have a duplicate, we'll just have to use a random uid.
+			dedupedName = titleDedupSet.deduplicate(name)
+		}
+		l.Debug("Duplicate alert rule name detected, renaming", "oldName", da.Name, "newName", dedupedName)
 		name = dedupedName
 	}
 	titleDedupSet.add(name)
 
-	// Here we ensure that the alert rule group is unique within the folder.
-	// This is so that we don't have to ensure that the alerts rules have the same interval.
-	groupDedupSet := om.AlertTitleDeduplicator(info.NewFolderUID)
-	panelSuffix := fmt.Sprintf(" - %d", da.PanelID)
+	// We need to ensure that all rules in a group have the same interval but would also like the group to contain useful information.
+	// To do this, we construct the group name from the dashboard title and the interval.
+	interval := ruleAdjustInterval(da.Frequency)
+	panelSuffix := fmt.Sprintf(" - %ds", interval)
```

Comment:
what if we humanize duration? that would make 3600s be a shorter '1hr

Reply:
[a415893](https://github.com/grafana/grafana/pull/78351/commits/a415893727714702341ae44d93925ac2d40e206d)